### PR TITLE
add functionality to add and save property constraints

### PR DESCRIPTION
**Pull request by TStadelmaier**
add functionality to create property constraints:
 - frontend: add input field to enter xml
 - backend: add functionality to add constraint as xml

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/TEntityTemplateResource.java [hunk 1]
```diff
@@ -41,4 +45,13 @@ public PropertiesResource getPropertiesResource() {
         return new PropertiesResource(this.o, (AbstractComponentInstanceResource) this.res);
     }
 
+    @GET
+    @Path("/propertyconstraints")
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/TEntityTemplateResource.java]: leading slash is not needed

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java [hunk 2]
```diff
@@ -141,6 +143,16 @@ public Response getReferenceCount(
         }
     }
 
+    @Path("propertyconstraints/")
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java]: No trailing slash either please.

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/policytemplates/PolicyTemplateResource.java [hunk 3]
```diff
@@ -71,4 +74,13 @@ public Response setName(String name) {
         return RestUtils.persist(this);
     }
 
+    @Path("propertyconstraints/")
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/policytemplates/PolicyTemplateResource.java]: remove trailing slash please

**Comment by lharzenetter** [comment]
Please consider the issue outlined by Codacy and resolve the conflicts with the master branch.

**Pull request merged**